src/Send/AssetSelectorScreen/AssetSelectorScreen.tsx

```diff
@@ -10,28 +11,40 @@ import NoImage from '../../assets/img/asset_no_image.png'
 import {Boundary, Button, Spacer, Text, TextInput} from '../../components'
 import {useTokenInfo} from '../../hooks'
 import globalMessages, {txLabels} from '../../i18n/global-messages'
+import {getDefaultAssetByNetworkId} from '../../legacy/config'
 import {decodeHexAscii, formatTokenAmount, getAssetDenominationOrId, getTokenFingerprint} from '../../legacy/format'
 import {useSelectedWallet} from '../../SelectedWallet'
 import {COLORS} from '../../theme'
-import {Token, TokenEntry} from '../../types'
+import {Token} from '../../types'
+import {YoroiWallet} from '../../yoroi-wallets'
+import {Quantity, TokenId, YoroiAmounts} from '../../yoroi-wallets/types'
 
 type Props = {
-  assetTokens: Array<TokenEntry>
-  onSelect: (tokenEntry: TokenEntry) => void
+  balance: YoroiAmounts
+  onSelect: (tokenId: TokenId) => void
   onSelectAll: () => void
 }
 
-export const AssetSelectorScreen = ({assetTokens, onSelect, onSelectAll}: Props) => {
+export const AssetSelectorScreen = ({balance, onSelect, onSelectAll}: Props) => {
   const strings = useStrings()
-
+  const wallet = useSelectedWallet()
+  const defaultAsset = getDefaultAssetByNetworkId(wallet.networkId)
   const [matcher, setMatcher] = React.useState('')
   const onChangeMatcher = (matcher: string) => {
     LayoutAnimation.configureNext(LayoutAnimation.Presets.easeInEaseOut)
     setMatcher(matcher)
   }
-  const sortedAssetTokens = assetTokens
-    .sort((a: TokenEntry, b: TokenEntry) => (a.amount.isGreaterThan(b.amount) ? -1 : 1))
-    .sort((assetToken) => (assetToken.identifier === '' ? -1 : 1))
+
+  const sortedBalance: Array<[TokenId, Quantity]> = useMemo(
+    () =>
+      Object.entries(balance)
+        .sort(
+          ([, quantityA]: [TokenId, Quantity], [, quantityB]: [TokenId, Quantity]) =>
+            parseInt(quantityB) - parseInt(quantityA),
```

Comment:
parseInt is limited to 1^53-1, quantity here is a string coming from a u64 bignumber
```suggestion
            new BN(quantityB).minus(new BN(quantityA)),
```

Reply:
Thank you. I have applied the same method as before using the BigNumber method `isGreaterThan`, so we dont need extra transformation. Is it right for you?